### PR TITLE
md fixes

### PR DESCRIPTION
**Pull request by chris-ha458**
Strips should_strip_sig_or_bom
some blackmagic boolean logic 4899bf3

I doubled checked this with pen and paper, and tests all pass
But still it would be a good idea to get another set of eyes here.

### CODE REVIEW

**Comment by chris-ha458** [comment]
i kept the clippy error messages.
If it is true that atleast in rust that `should_strip_sig_or_bom` always returns true,
strip_sig_or_bom is always equal to bom_or_sig_available.
This means that `bom_or_sig_available && !strip_sig_or_bom` becomes
`bom_or_sig_available && !bom_or_sig_available ` which is always false.
this occurs twice and clippy did not find them before but could find them with these commits.

File: src/lib.rs [hunk 1]
```diff
@@ -446,7 +446,8 @@ pub fn from_bytes(bytes: &[u8], settings: Option<NormalizerSettings>) -> Charset
             if md_ratios.last().unwrap() >= &settings.threshold {
                 early_stop_count += 1;
             }
-            if early_stop_count >= max_chunk_gave_up || (bom_or_sig_available && !strip_sig_or_bom)
+            if early_stop_count >= max_chunk_gave_up
+                || (bom_or_sig_available && !bom_or_sig_available)
```
Comment (nickspring) [inline src/lib.rs:450]: can be removed?

Comment (nickspring) [inline src/lib.rs:450]: can be removed?

Comment (chris-ha458) [reply src/lib.rs:450]: Yes my opinion is that this block, and other `(bom_or_sig_available && !bom_or_sig_available)` can be all removed or modified accordingly.
It effects many parts of the code so I left it in for double checking.
As it stands clippy does complain and mentions all occurences of it.

Comment (chris-ha458) [reply src/lib.rs:450]: If you agree I can push a commit that does those changes.

Comment (nickspring) [reply src/lib.rs:450]: yes, please push it

**Pull request merged**